wdl-ast/src/lib.rs

```diff
@@ -96,6 +97,62 @@ impl<T: AstNode> AstNodeExt for T {
     }
 }
 
+/// An extension trait for syntax nodes.
+pub trait SyntaxNodeExt {
```

Comment:
I think this is how I initially wrote it and you just followed my lead, but in hindsight I think this might be better suited as a `SyntaxElementExt`?

How things are set up in `wdl-format` is that comments are "owned" by other tokens (never nodes). So it seems strange to treat "except comments" here as being owned solely by nodes and not tokens. It might be more consistent to implement this extension trait for `NodeOrToken`.

But that doesn't need to happen in this PR. Just something to think about and maybe change in the future.